lib/posix-environ/environ.c

```diff
@@ -183,3 +186,44 @@ char **__environ = __init_env;
 __weak_alias(__environ, ___environ);
 __weak_alias(__environ, _environ);
 __weak_alias(__environ, environ);
+
+/* Find the first occurrence of a variable within __init_env */
+static char **find_vnpos(const char *varname, size_t varname_len, char **end)
+{
+	char **i;
+
+	for (i = __init_env; *i && i != end; i++) {
+		if (!strncmp(varname, *i, varname_len) &&
+		    (varname_len[*i] == '=' || varname_len[*i] == '\0'))
+			return i;
+	}
+	return NULL;
+}
+
+/* Eliminate duplicates from __init_env */
+static int uniquify_env(struct uk_init_ctx *ictx __unused)
+{
+	char **rpos; /*< read position */
+	char **wpos; /*< write position */
+
+	for (rpos = __init_env, wpos = __init_env; *rpos; rpos++) {
+		size_t namelen;
+		char **opos; /*< overwrite position */
+
+		namelen = strchrnul(*rpos, '=') - *rpos;
```

Comment:
Since environment variables are `key - value` pairs, would it be a good idea to instead throw an assert here in case `strchrnul` did return a char pointer to `\0`? In that case maybe `strchr` and checking for `NULL` would suffice.

Reply:
I am matching the behavior of musl here (actually the ported functions, like `getenv`). You can find the same length computation in `getenv`, so I think this code should do the same assumption: an environment variable could be just the `key` without value.
I think an assertion is even worse because variables can be set from user input (e.g., `env.vars=` kernel parameter). If we would compile production builds, the assertion would be optimized out...